code/modules/library/admin_only.dm

```diff
@@ -0,0 +1,354 @@
+
+	var/datum/db_query/query_update_log = SSdbcore.NewQuery({"
+		INSERT INTO [format_table_name("library_action")] (book, reason, ckey, datetime, action, ip_addr)
+		VALUES (:book, :reason, :ckey, Now(), :action, INET_ATON(:ip_addr))
+	"}, list("book" = id, "reason" = reason, "ckey" = admin.ckey, "action" = BOOK_ADMIN_DELETE, "ip_addr" = admin.address))
+	query_update_log.warn_execute()
+	qdel(query_update_log)
+	library_updated()
+	params_changed = TRUE
+	update_db_info()
+
+/obj/machinery/computer/libraryconsole/admin_only_do_not_map_in_you_fucker/proc/unhide_book(id, reason, client/admin)
+	if(!SSdbcore.Connect())
+		can_connect = FALSE
+		message_admins("Could not connect to the DB")
+		return
+	if(IsAdminAdvancedProcCall())
+		stack_trace("Proc call lead to unhide_book being called, blocked")
+		message_admins("Proc call lead to unhide_book being called, blocked")
+		return
+	if(!check_rights_for(admin, R_BAN))
+		stack_trace("[admin.ckey] tried to unhide a book without the required perms")
+		return
+
+	log_admin_private("([admin.ckey]) unhid book #[id]")
+
+	var/datum/db_query/query_unhide_book = SSdbcore.NewQuery({"
+		UPDATE [format_table_name("library")]
+		SET deleted = NULL
+		WHERE id = :id
+	"}, list("id" = id))
+
+	query_unhide_book.warn_execute()
+	qdel(query_unhide_book)
+
+	var/datum/db_query/query_update_log = SSdbcore.NewQuery({"
+		INSERT INTO [format_table_name("library_action")] (book, reason, ckey, datetime, action, ip_addr)
+		VALUES (:book, :reason, :ckey, Now(), :action, INET_ATON(:ip_addr))
+	"}, list("book" = id, "reason" = reason, "ckey" = admin.ckey, "action" = BOOK_ADMIN_RESTORE, "ip_addr" = admin.address))
+	query_update_log.warn_execute()
+	qdel(query_update_log)
+	library_updated()
+	update_db_info()
+
+/// This mostly exists to document the form of the library_action table, since it doesn't do that good a job on its own
+/datum/book_history_entry
+	/// The id of this logged action
+	var/id
+	/// The book id this log applies to
+	var/book
+	/// The reason this action was enacted
+	var/reason
+	/// The admin who performed the action
+	var/ckey
+	/// The time of the action being performed
+	var/datetime
+	/// The action that occured (BOOK_ADMIN_DELETE, BOOK_ADMIN_RESTORE, and legacy BOOK_ADMIN_REPORT)
+	var/action
+	/// The ip address of the admin who performed the action
+	var/ip_addr
+
+/datum/book_history_entry/proc/serialize()
+	var/list/data = list()
+	data["id"] = id
+	data["book"] = book
+	data["reason"] = reason
+	data["ckey"] = ckey
+	data["datetime"] = datetime
+	data["action"] = action
+	data["address"] = ip_addr
+	return data
+
+/datum/admin_book_viewer
+	/// Weakref to the /obj/machinery/computer/libraryconsole/admin_only_do_not_map_in_you_fucker that spawned us
+	var/datum/weakref/owner_ref
+	var/view_raw = FALSE
+	var/id
+	var/author
+	var/title
+	var/content
+	var/category
+	var/author_ckey
+	var/creation_time
+	var/deleted
+	var/creation_round
+	var/list/datum/book_history_entry/history
```

Comment:
you dmdocced everything else... why not this?

Reply:
I thought it was decently self explanatory. can do so if you'd like

Reply:
Just wondering if it was a forgetting thing or not. The only reason for it to be valuable is to ensure our dmdoc stats stay good so it's ultimately up to you

Reply:
I'll doc it, sure